webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -0,0 +1,291 @@
+import { DownloadOutlined, FileOutlined, LeftOutlined } from '@ant-design/icons';
+import { Tooltip, Tree } from 'antd';
+import { DataNode } from 'antd/lib/tree';
+import yaml from 'js-yaml';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import MonacoEditor from 'components/MonacoEditor';
+import Section from 'components/Section';
+import useResize from 'hooks/useResize';
+import { handlePath, paths } from 'routes/utils';
+import { getExperimentFileFromTree, getExperimentFileTree } from 'services/api';
+import { V1FileNode as FileNode } from 'services/api-ts-sdk';
+import Spinner from 'shared/components/Spinner';
+import { RawJson } from 'shared/types';
+import { isEqual } from 'shared/utils/data';
+import { ErrorType } from 'shared/utils/error';
+import handleError from 'utils/error';
+
+const { DirectoryTree } = Tree;
+
+import css from './CodeViewer.module.scss';
+import './index.scss';
+
+export type Props = {
+  configRaw?: RawJson;
+  experimentId: number;
+}
+
+type FileInfo = {
+  data: string;
+  name: string;
+  path: string;
+};
+
+/**
+ * A component responsible to enable the user to view the code for a experiment.
+ *
+ * It renders a file tree and a selected file in the MonacoEditor
+ *
+ * Props:
+ *
+ * experimentID: the experiment ID;
+ *
+ * configRaw: the experiment.configRaw property to be used to render a Config yaml file;
+ */
+const CodeViewer: React.FC<Props> = ({ experimentId, configRaw }) => {
+  const resize = useResize();
+
+  const [ config ] = useState(() => {
```

Comment:
quesrion:  does having this as a `setState` do for you?

Reply:
>quesrion: does having this as a setState do for you?
sorry, mean to ask "what does having this as a setState do for you?"

>I though that it would be appropriate due to the fact that it depends on a prop
but the expression inside of `useState` is only evaluated on the initial render, when the prop changes, `config` will not be updated.